README.md

```diff
@@ -2787,38 +2804,52 @@ format = "{data!j}"
 
+#### Nested JSON
 
-### Transformation data
+Within templates and formats, you can refer only to the top-level names of an incoming JSON message,
+which significantly limits the kinds of messages `mqttwarn` can process. A [solution is in the works](https://github.com/jpmens/mqttwarn/issues/303)
+for this, but in the meantime you can use an `alldata` function to transform the JSON into something 
+`mqttwarn` _can_ process.
 
-_mqttwarn_ can transform an incoming message before passing it to a plugin service.
-On each message, _mqttwarn_ attempts to decode the incoming payload from JSON. If
-this is possible, a dict with _transformation data_ is made available to the
-service plugins as `item.data`.
+The trick is to build a new JSON message with _only_ top-level values, specifically the values you need.
 
-This transformation data is initially set up with some built-in values, in addition
-to those decoded from the incoming JSON payload. The following built-in variables
-are defined in `item.data`:
+### Using custom functions
 
-```python
-{
-  'topic'         : topic name
-  'payload'       : topic payload
-  '_dtepoch'      : epoch time                  # 1392628581
-  '_dtiso'        : ISO date (UTC)              # 2014-02-17T10:38:43.910691Z
-  '_dthhmm'       : timestamp HH:MM (local)     # 10:16
-  '_dthhmmss'     : timestamp HH:MM:SS (local)  # 10:16:21
-}
-```
+A topic section in the INI file can have properties set as per the table at the bottom of [this section](https://github.com/jpmens/mqttwarn#the-__topic__-sections). The `targets`, `topic` and `qos` properties can not be defined with a function. 
 
-Any of these values can be used in `format` to create custom outgoing
-messages.
+#### Topic-section properties that can call a custom function
 
-```ini
-format = I'll have a {fruit} if it costs {price} at {_dthhmm}
-```
+- `datamap` : dictionary, or a function that returns a dictionary
+- `alldata` : dictionary, or a function that returns a dictionary
+- `filter` : boolean, or a function that returns a boolean
+- `title` : string, or a function that returns a string
+- `format` : string, or a function that returns a string
+- `priority` : see below
+- `image` : see below
+
+#### Data mapping functions
+
+Both the `datamap` and the `alldata` properties in a topic section can call a function which returns a [dictionary](https://docs.python.org/2/tutorial/datastructures.html#dictionaries). The keys in this dictionary can be used when describing the outbound `title` and `format` properties of the same topic section.
+
+A function called using the `datamap` property, gets access to the `topic` string and the `service` object. Functions called using the `alldata` property, get access to the `data` dictionary in addition to that. It's not clear to me why both of these exist, as `alldata` seems to be the more extensive environment to use. 
+
+- `topic`: again, seems superfluous, as `data['topic']` contains the same value
```

Comment:
true enough, but not appropriate for the README?

Reply:
Agreed. Again, I copied and pasted the text out of a comment I made earlier and forgot to edit this out. I've left the "topic" mention in for now, because of the above comment.